File: Utilities/Models/Actor.cs



namespace Utilities.Models;

public class Actor : BaseModel
{
    [Required] 
    public string LastName { get; set;}
    [Required]
    public int? Age { get; set; }
    [JsonIgnore]
    public ICollection<Movie>? Movies { get; set; } = new List<Movie>();
}

File: Client/src/app/Models/actor.ts
import { BaseModel, IBaseModel } from './baseModel';
import { IMovie } from './movie';

export interface IActor extends IBaseModel {
  age?: number;
  movies?: IMovie[];
}

export class Actor extends BaseModel implements IActor {
  age?: number;
  movies?: IMovie[];

  constructor() {
    super();
    this.movies = [];
  }
}
